src/backend/distributed/utils/aggregate_utils.c

```diff
@@ -703,20 +791,65 @@ TypecheckWorkerPartialAggArgType(FunctionCallInfo fcinfo, StypeBox *box)
 	Assert(!argtypesNull);
 	ReleaseSysCache(proctuple);
 
-	if (ARR_NDIM(DatumGetArrayTypeP(argtypes)) != 1 ||
-		ARR_DIMS(DatumGetArrayTypeP(argtypes))[0] != 1)
+	if (ARR_NDIM(DatumGetArrayTypeP(argtypes)) != 1)
 	{
 		elog(ERROR, "worker_partial_agg_sfunc cannot type check aggregates "
-					"taking anything other than 1 argument");
+					"taking multi-dimensional arguments");
 	}
 
-	int arrayIndex = 0;
-	Datum argtype = array_get_element(argtypes,
-									  1, &arrayIndex, -1, sizeof(Oid), true, 'i',
-									  &argtypesNull);
-	Assert(!argtypesNull);
+	int naggarg = ARR_DIMS(DatumGetArrayTypeP(argtypes))[0];
+
+	/* we expect aggregate function to have at least a single parameter */
+	if (naggarg == 0)
+	{
+		return false;
+	}
+
+	int idx = 0;
+	Datum argtype;
```

Comment:
Minor convention note: `argtype` -> `argType`

Reply:
Will change. As a note, I tried to name parameters in a way similar to surrounding variables. For instance, there is an existing variable "argtypes" in this function.

Reply:
Yeah, this function already misses few of our conventions I guess. For example, I'd  have preferred   `argtypes ` -> `argumentTypes`.

(These are not very strictly enforced, we can live with some exceptions. As this is your first PR I wanted to explicitly bring all of them up :) )